walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/servlets/FindMeetingQuery.java

```diff
@@ -0,0 +1,158 @@
+                    else
+                    {
+                        checkOverlapsInPotentialTimesWithEventTimes(availableTime, whenEvent, meetingDuration);   
+                    }
+                }  
+                else if (request.getOptionalAttendees().isEmpty())
+                {
+                    return Arrays.asList(TimeRange.WHOLE_DAY);
+                } 
+                //Checks if optional attendees can attend open times, but will not affect times made for mandatory attendees
+                if (checkAttendees(eventAttendees, request.getOptionalAttendees()) && !request.getAttendees().isEmpty())
+                {
+                    return checkIfOptionalAttendeesCanAttend(availableTime,whenEvent);
+                }
+            }
+        }
+        return availableTime;
+    }
+
+    public void addFirstOpenTimes(long meetingDuration, Collection<TimeRange> availableTime, TimeRange whenEvent)
+    {
+        TimeRange frontTime1= TimeRange.fromStartEnd(TimeRange.START_OF_DAY,whenEvent.start(),false);
+        TimeRange backTime1 = TimeRange.fromStartEnd(whenEvent.end(),TimeRange.END_OF_DAY,true);
+        addFrontAndBackTimesToList(meetingDuration, frontTime1, backTime1, availableTime);
+    }
+    
+    public void checkOverlapsInPotentialTimesWithEventTimes(Collection<TimeRange> availableTime,TimeRange whenEvent,
+        long meetingDuration)
+    {
+        Collection<TimeRange> removeList = new ArrayList<TimeRange>();
+        Collection<TimeRange> addList = new ArrayList<TimeRange>();
+        for (TimeRange openTime : availableTime)
+        {
+            //check if times overlap to produce the right time slots around the current event
+            if(openTime.overlaps(whenEvent))
+            {
+                if(openTime.contains(whenEvent))
+                {
+                    removeList.add(openTime);
+                    TimeRange frontTime2 = TimeRange.fromStartEnd(openTime.start(),whenEvent.start(),false);
+                    TimeRange backTime2 = TimeRange.fromStartEnd(whenEvent.end(),TimeRange.END_OF_DAY,true);
+                    addFrontAndBackTimesToList(meetingDuration, frontTime2, backTime2,addList);
+                }
+                else if (whenEvent.start() < openTime.start())
+                {
+                    removeList.add(openTime);
+                    TimeRange overlapTime = TimeRange.fromStartEnd(whenEvent.end(),openTime.end(),false);
+                    if (overlapTime.duration() >= meetingDuration)
+                    {
+                        addList.add(overlapTime);
+                    }
+                }
+            }
+        }
+        availableTime.removeAll(removeList);
+        availableTime.addAll(addList);    
+    }
+
+    public Collection<TimeRange> checkIfOptionalAttendeesCanAttend(Collection<TimeRange> availableTime,TimeRange whenEvent)
+    {
+        Collection<TimeRange> timeWithOptional = new ArrayList<TimeRange>(); 
+        int count = 0;
```

Comment:
since you don't actually use the `count` as an `int`, you can delete it and just check if `timeWithOptional.empty()` below.